src/components/Queue.vue

```diff
@@ -113,68 +128,77 @@ export default {
 
             setTimeout(function() {
                 this.validate = false;
-                this.message = '';
             }.bind(this), RESULT_TIMER);
         },
 
         showFail: function() {
             this.spinner = false;
             this.fail = true;
 
-            this.message = "Whoops! There may be a network problem. When you have a better signal, click 'Submit queued vouchers' to retry.";
             setTimeout(function() {
                 this.fail = false;
-                this.message = '';
             }.bind(this), RESULT_TIMER);
         },
 
+        genQueueSuccessMessage: function(response) {
+            var data = response.data;
+            var success = '';
+            var fail = '';
+            var invalid = '';
+
+            // Construct the feedback message.
+            if (data.success.length === 1) {
+                success = "1 voucher was accepted, ";
+            } else {
+                success = data.success.length + " vouchers were accepted, ";
+            }
+
+            if (data.fail.length === 1) {
+                fail = " 1 was a duplicate ";
+            } else {
+                fail = data.fail.length + " were duplicates ";
+            }
+
+            if (data.invalid.length === 1) {
+                invalid = "and 1 was invalid.";
+            } else {
+                invalid = "and " + data.invalid.length + " were invalid.";
+            }
+
+            var message
+                = "Thanks! Your queue has been successfully submitted. "
```

Comment:
I just took the message as I found it. It's not related directly to the message component so it should be resolved in another card most likely.